eth/stagedsync/stage_headers.go

```diff
@@ -566,13 +566,24 @@ func verifyAndSaveNewPoSHeader(
 		// Side chain or something weird
 		// TODO(yperbasis): considered non-canonical because some missing headers were downloaded but not canonized
 		// Or it's not a problem because forkChoice is updated frequently?
-
+		if cfg.memoryOverlay {
+			var checked bool
+			if checked, err = cfg.hd.ValidatePayload(tx, header, body, false, cfg.execPayload); err != nil {
+				return &privateapi.PayloadStatus{Status: remote.EngineStatus_INVALID}, false, nil
+			}
+			if checked {
+				return &privateapi.PayloadStatus{
+					Status:          remote.EngineStatus_VALID,
+					LatestValidHash: currentHeadHash,
+				}, true, nil
+			}
+		}
 		// No canonization, HeadHeaderHash & StageProgress are not updated
 		return &privateapi.PayloadStatus{Status: remote.EngineStatus_ACCEPTED}, true, nil
 	}
 
 	if cfg.memoryOverlay && (cfg.hd.GetNextForkHash() == (common.Hash{}) || header.ParentHash == cfg.hd.GetNextForkHash()) {
```

Comment:
What's the point of using in-memory validation when `currentHeadHash == header.ParentHash`?

Reply:
we optimize the rpcdaemon, we are supposed to actually commit on fcu